src/mapping.ts

```diff
@@ -59,6 +63,66 @@ export function handleCreateProxy(event: Created): void {
   proxy.save()
 }
 
+function loadOrCreatePool(poolMeta: PoolMeta, block: EthereumBlock): Pool {
+  let pool = Pool.load(poolMeta.id)
+
+  log.debug("pool start block {}, current block {}", [poolMeta.startBlock.toString(), block.number.toString()])
+  if (pool == null && parseFloat(block.number.toString()) >= poolMeta.startBlock) {
+    createPool(poolMeta.id.toString())
+    pool = Pool.load(poolMeta.id)
+  }
+
+  log.debug("successfully using this for pool meta id: {}", [poolMeta.id.toString()])
+  return <Pool>pool    
+}
+
+function createYesterdaySnapshot(date: BigInt, block: EthereumBlock): void {
+  let yesterdayTimeStamp = date.minus(BigInt.fromI32(86400))
+  let yesterday = Day.load(yesterdayTimeStamp.toString())
+
+  let relevantPoolMetas = poolMetas.filter(poolMeta => poolMeta.networkId == dataSource.network())
+  for (let i = 0; i < relevantPoolMetas.length; i++) {
+    let poolMeta = relevantPoolMetas[i]
+
+    let pool = loadOrCreatePool(poolMeta, block)
+    if (pool == null) {
+      continue
+    }
+
+    let dailyPoolData = createDailyPoolData(poolMeta, yesterday.id)
+
+    // only get these values in v3
+    if (poolMeta.version == 3) {
+      let reserveContract = Reserve.bind(<Address>Address.fromHexString(poolMeta.reserve))
+      let reserve = reserveContract.totalBalance()
+      dailyPoolData.reserve = reserve
+
+      let navFeedContract = NavFeed.bind(<Address>Address.fromHexString(poolMeta.nftFeed))
+      let currentNav = navFeedContract.currentNAV()
+      dailyPoolData.assetValue = currentNav
+    } 
+
+    dailyPoolData.totalDebt = pool.totalDebt
+    dailyPoolData.seniorDebt = pool.seniorDebt
+    dailyPoolData.currentJuniorRatio = pool.currentJuniorRatio
+    dailyPoolData.save()
+  }
+}
+
+// capture a snapshot of the investments by day
+function runRewardLogic(block: EthereumBlock): void {
```

Comment:
This isn't really related to the rewards right? Maybe this function should have a different name, e.g. `createDailySnapshot()`?